pkg/functiontest/testing_test.go

```diff
@@ -0,0 +1,77 @@
+package functiontest
+
+import (
+	"sync"
+	"testing"
+
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func TestEvaluateBasics(t *testing.T) {
+	fn := func(inputs struct{}) ([]client.Object, error) {
+		output := &corev1.Pod{}
+		output.Name = "test-pod"
+		return []client.Object{output}, nil
+	}
+
+	Evaluate(t, fn, Scenario[struct{}]{
+		Name:   "example-test",
+		Inputs: struct{}{},
+		Assertion: func(t *testing.T, scen *Scenario[struct{}], outputs []client.Object) {
+			t.Logf("asserting on %d outputs", len(outputs))
+		},
+	})
+}
+
+func TestAssertionChain(t *testing.T) {
+	fn := func(inputs struct{}) ([]client.Object, error) {
+		output := &corev1.Pod{}
+		output.Name = "test-pod"
+		return []client.Object{output}, nil
+	}
+
+	Evaluate(t, fn, Scenario[struct{}]{
+		Name:   "example-test",
+		Inputs: struct{}{},
+		Assertion: AssertionChain(
+			func(t *testing.T, scen *Scenario[struct{}], outputs []client.Object) {
```

Comment:
Should we assert that these were actually called? maybe an atomic call counter and check the count after evaluate is done running?